libs/grok/src/main/java/org/elasticsearch/grok/PatternBank.java

```diff
@@ -112,25 +121,22 @@ private static void innerForbidCircularReferences(Map<String, String> bank, Stri
                 end = bracketIndex;
             } else if (columnIndex != -1 && bracketIndex == -1) {
                 end = columnIndex;
-            } else if (bracketIndex != -1 && columnIndex != -1) {
+            } else if (bracketIndex != -1) {
```

Comment:
For future GitHub archaeologists: this change looks out of place but isn't, IntelliJ warns that `columnIndex != -1` is always true here, so it's safe to have removed this.